app/services/dttp/update_trainee_status.rb

```diff
@@ -31,10 +32,12 @@ def call
 
   private
 
-    attr_reader :path, :params, :entity_id
+    attr_reader :path, :params, :trainee, :entity_type
 
-    def trainee
-      Trainee.find_by(dttp_id: entity_id)
+    def entity_id
```

Comment:
Happy to go along with this change. However, if one were to seek minimal changes, you could do something like this. What do you think?

```ruby
DTTP_COLUMN_NAME = {
  CONTACT_ENTITY_TYPE: :dttp_id,
  PLACEMENT_ASSIGNMENT_ENTITY_TYPE: :placement_assignment_dttp_id,
}.freeze

...

def trainee
  Trainee.find_by(DTTP_COLUMN_NAME[entity_type] => entity_id)
end
```

Reply:
We opted to get rid of the `find_by` call since it could return a `nil` in some cases. Since all calls to `Dttp::UpdateTraineeStatus` were previously passing an attribute off `trainee`, it made sense to send the whole object instead and let the service decide the endpoint to talk to.